File: evidence/generated-systems/angular/analytics-on-angular/src/app/models/ModelVersion.js

// Define collection and schema for ModelVersion
export interface ModelVersion {
    version:
	type : string
    Model_:
	type : Schema.Types.ObjectId
    TrainingRun:
	type : Schema.Types.ObjectId
    EvaluationMetrics:
 	type : [{ type: Schema.Types.ObjectId, ref: 'EvaluationMetric' }]
    Deployments:
 	type : [{ type: Schema.Types.ObjectId, ref: 'InferenceEndpoint' }]
    FeatureSets:
 	type : [{ type: Schema.Types.ObjectId, ref: 'FeatureSet' }]
    Datasets:
 	type : [{ type: Schema.Types.ObjectId, ref: 'DataSet' }]
    Lifecycle:
 	type : String
    TrainingStatus:
 	type : String
#
    collection: 'modelVersions'
}

File: evidence/generated-systems/golang/analytics-on-golang/analytics-on-golang/internal/model/ModelVersion.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// ModelVersion Declaration
//==============================================================
type ModelVersion struct {
    gorm.Model
     Version                                    string
    Model_Id         *uint
    Model_           *Model_ `gorm:"foreignKey:Model_Id"`
    TrainingRunId         *uint
    TrainingRun           *TrainingRun `gorm:"foreignKey:TrainingRunId"`
     EvaluationMetrics           []EvaluationMetric `gorm:"foreignKey:EvaluationMetricsFromModelVersionId"`
     Deployments           []InferenceEndpoint `gorm:"foreignKey:DeploymentsFromModelVersionId"`
     FeatureSets           []FeatureSet `gorm:"foreignKey:FeatureSetsFromModelVersionId"`
     Datasets           []DataSet `gorm:"foreignKey:DatasetsFromModelVersionId"`
    Lifecycle                      ModelLifecycle
    TrainingStatus                      TrainingStatus

// parent associations as their child

}

